parser/src/lib.rs

```diff
@@ -50,6 +50,24 @@ pub enum Fragment<'f> {
     Parameter(Parameter),
 }
 
+/// A parsed formatting parameter (contents of `{` `}` block).
+///
+/// # Syntax
+///
+/// ```notrust
+/// param := '{' [ argument ] [ '=' argtype ] [ ':' format_spec ] '}'
+/// argument := integer
+///
+/// argtype := bitfield | '?' | format-array | '[?]' | byte-array | '[u8]' | 'istr' | 'str' |
+///     'bool' | 'char' | 'u8' | 'u16' | 'u32' | 'u64' | 'u128' | 'usize' | 'i8' | 'i16' | 'i32' |
+///     'i64' | 'i128 | 'isize' | 'f32'
+/// bitfield := integer '..' integer
+/// format-array := '[?;' ' '* integer ']'
```

Comment:
```suggestion
/// format-array := '[?;' ' ' * integer ']'
```

Reply:
Hmm, these are supposed to be [Kleene stars](https://en.wikipedia.org/wiki/Kleene_star), to denote zero or more repetitions of `' '` are allowed, so they shouldn't be spaced apart. Would it be clearer if I moved the `' '*` to a `spaces := ' '*` rule?

Reply:
ugh duh, shouldn't review stuff like this before my first coffee :D Yeah, the reason for my suggestion was that I had a hard time figuring out what the "opening" and "closing" ticks were. a `spaces := ' '*` rule would make this clearer imo.

Reply:
Okay, did that